snapcraft/internal/project_loader/_config.py

```diff
@@ -108,6 +108,13 @@ def __init__(self, project_options=None):
         _ensure_grade_default(snapcraft_yaml, self._validator.schema)
 
         self.data = self._expand_env(snapcraft_yaml)
+        self._project_options.info = snapcraft.ProjectInfo(
```

Comment:
Quick note based on the PR description: eventually the CLI should be passing in a `ProjectEnvironment` instance and the `Project` can be assembled here, removing the need to set attributes.

Reply:
I'll add a comment.